src/collections/point_collection.rs

```diff
@@ -119,15 +119,15 @@ impl PointCollection {
     /// assert_eq!(pc.len(), 2);
     /// ```
     pub fn from_data(
-        coordinates: Vec<Vec<Coordinate>>,
+        coordinates: Vec<Vec<Coordinate2D>>,
         time_intervals: Vec<TimeInterval>,
         data: HashMap<String, FeatureData>,
     ) -> Result<Self> {
         let capacity = coordinates.len();
 
-        let mut fields = vec![
-            Field::new(Self::FEATURE_FIELD, Self::multi_points_data_type(), false),
-            Field::new(Self::TIME_FIELD, Self::time_data_type(), false),
+        let mut columns = vec![
+            Field::new(Self::FEATURE_COLUMN, Self::multi_points_data_type(), false),
+            Field::new(Self::TIME_COLUMN, Self::time_data_type(), false),
         ];
 
         let mut builders: Vec<Box<dyn ArrayBuilder>> = vec![
```

Comment:
The multi-points and time intervals are not PrimitiveTypes, they are Lists and fixed sized lists. This way, for other applications, we have something like this: `<f64, f64>, ..., <f64, f64>` rather than arrays of binary data. Moreover, multi-points are variable-sized lists of these 2-tuples.